### PR TITLE
Challenge1

### PR DESCRIPTION
**Pull request by sambiase**
New PR

### CODE REVIEW

File: src/main/java/mainClass.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class mainClass {
+    public static void main(String[] args) {
+
+        csvManipulator csv = new csvManipulator();
+        final String READ_PATH = args[0];
+        final String WRITE_PATH = args[1];
+        final List<String> INPUT_DATA = new ArrayList<>();
```
Comment (jcralmeida) [inline src/main/java/mainClass.java:10]: Variable names in Java follow the camel case pattern:

e.g;

readPath, writePath

File: src/main/java/old/readCSV.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package old;
+
```
Comment (jcralmeida) [inline src/main/java/old/readCSV.java:2]: You can remove this old implementations

File: src/main/java/csvManipulator.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    // WRITE CSV
+    public void csvWrite(String write_path, List <String> inputData){
+        try {
+            CSVPrinter writeFile = new CSVPrinter(new FileWriter(write_path),CSVFormat.DEFAULT);
+            for (String input:inputData){
+                writeFile.printRecord(input);
+            }
+            writeFile.flush();
+            writeFile.close();
```
Comment (jcralmeida) [inline src/main/java/csvManipulator.java:36]: Take a look at the try-with-resources, if you use it, this close will not be necessary

File: src/main/java/csvManipulator.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+public class csvManipulator {
+
+    // READ CSV
+    public void csvRead(String read_path){
+        try{
+            CSVParser csvRead = new CSVParser(new FileReader(read_path), CSVFormat.DEFAULT.withHeader("id","first_name","last_name"));
+            for (CSVRecord record: csvRead) {
+                System.out.println("ID:"+record.get("id")+" - Full Name: "+record.get("first_name") + " "+ record.get("last_name"));
+            }
+            csvRead.close();
```
[the remark this answers is not in the document]
Comment (jcralmeida) [reply src/main/java/csvManipulator.java:20]: https://www.baeldung.com/java-try-with-resources

File: src/main/java/csvManipulator.java (hunk 3 above)
Comment (jcralmeida) [reply src/main/java/csvManipulator.java:36]: https://www.baeldung.com/java-try-with-resources

File: src/main/java/csvManipulator.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            CSVPrinter writeFile = new CSVPrinter(new FileWriter(write_path),CSVFormat.DEFAULT);
+            for (String input:inputData){
+                writeFile.printRecord(input);
+            }
+            writeFile.flush();
+            writeFile.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
```
Comment (jcralmeida) [inline src/main/java/csvManipulator.java:40]: nit: remove extra empty line

File: src/main/java/csvManipulator.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
```
Comment (jcralmeida) [inline src/main/java/csvManipulator.java:42]: nit: remove extra empty line

File: src/main/java/csvManipulator.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+}
+
```
Comment (jcralmeida) [inline src/main/java/csvManipulator.java:44]: nit: remove extra empty line

File: src/main/java/mainClass.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+        INPUT_DATA.add(args[2]);
+        INPUT_DATA.add(args[3]);
+
+        csv.csvRead(READ_PATH);
+        csv.csvWrite(WRITE_PATH, INPUT_DATA);
+
```
Comment (jcralmeida) [inline src/main/java/mainClass.java:16]: nit: remove extra empty line

File: src/main/java/mainClass.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    }
+
```
Comment (jcralmeida) [inline src/main/java/mainClass.java:18]: nit: remove extra empty line

File: src/main/java/mainClass.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class mainClass {
+    public static void main(String[] args) {
+
```
Comment (jcralmeida) [inline src/main/java/mainClass.java:6]: nit: remove extra empty line

File: pom.xml [hunk 11]
```diff
@@ -8,10 +8,26 @@
...
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-csv</artifactId>
+            <version>1.8</version>
+        </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>RELEASE</version>
+            <scope>compile</scope>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (sambiase) [reply pom.xml]: removed

File: src/main/java/csvManipulator.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.csv.CSVRecord;
+
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+public class csvManipulator {
```
[the remark this answers is not in the document]
Comment (sambiase) [reply src/main/java/csvManipulator.java]: fixed

**Pull request closed without merging**